openapi-contracts/src/main/com/sinch/sdk/domains/conversation/models/dto/v1/AgentJoinedEventDto.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Conversation API | Sinch
+ * Send and receive messages globally over SMS, RCS, WhatsApp, Viber Business, Facebook messenger and other popular channels using the Sinch Conversation API.  The Conversation API endpoint uses built-in transcoding to give you the power of conversation across all supported channels and, if required, full control over channel specific features.
+ *
+ * The version of the OpenAPI document: 1.0
+ * Contact: support@sinch.com
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+package com.sinch.sdk.domains.conversation.models.dto.v1;
+
+import com.fasterxml.jackson.annotation.JsonFilter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import java.util.Objects;
+
+/** AgentJoinedEventDto */
+@JsonPropertyOrder({AgentJoinedEventDto.JSON_PROPERTY_AGENT_JOINED_EVENT})
+@JsonFilter("uninitializedFilter")
+@JsonInclude(value = JsonInclude.Include.CUSTOM)
+public class AgentJoinedEventDto {
```

Comment:
The object should have a property `agent` but I can't find it. Can you help me understand this object's structure?

Reply:
Not yet handled this part (Events?) but seems this is coming from actual OAS description.
```
    AppEvent:
      oneOf:
        - type: object
          title: agent_joined_event
          properties:
            agent_joined_event:
              $ref: "#/components/schemas/AgentJoinedEvent"
```
So I'm afraid here that generator tools is creating an endless loop with a class depending of itself 😞